service/k8sclient/k8sclient.go

```diff
@@ -746,7 +747,7 @@ func (c *K8sClient) CreatePSMDBCluster(ctx context.Context, params *PSMDBParams)
 		return errors.Wrap(err, "cannot create secret for PXC")
 	}
 
-	affinity := &psmdb.PodAffinity{}
+	affinity := new(psmdb.PodAffinity)
```

Comment:
what's the reason for this change? we usually don't use `new` in Go unless we really need to. So if there is no special need for this, I would suggest not using new so that we can keep declarations consistent throughout our codebases

Reply:
`bin/go-consistent -pedantic -exclude "tests" ./...` kept failing on this. I was thinking: What's the point of having tests that fail and we ignore them failing. So I changed it.

Reply:
`zero value ptr alloc: use new(T) for *T allocation` is an error one gets.

Reply:
https://github.com/percona-platform/dbaas-controller/pull/78

If it works now, I think we should remove `-exclude "tests"` and then update _tests_ (despite go-consistent complaining about non-testing code – it picks the most common / consistent variant, but we want consistency avoid `new`). But let's do that in a separate PR

Reply:
I will remove this usage of `new` then.